### PR TITLE
osx: install.sh improvements

### PR DESCRIPTION
**Pull request by LinusU**
This improves the installation script on OS X in the following ways:

- Install all necessary libs with the correct flags (using brew)
- Only installs mingw and wine if `TARGET=windows` (we now have native OS X support so I think this is desirable)
- Remove support for macports (it has already been removed from the readme in #2459 and I don't think that anyone uses it anymore)

This is extracted from #2401 which is waiting for me to fix the bottles. I think it's a good idea to merge this in the mean time so that people can get up and running more easily on OS X.

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -5,9 +5,7 @@ SDL2_TTF_PV=2.0.12
 
 cachedir=.cache
 liburl=https://openrct2.website/files/orctlibs.zip
-mkdir -p $cachedir
-
-echo $(uname)
```
Comment (LinusU) [inline install.sh]: This isn't removed, it's just moved down to line 140

File: install.sh [hunk 2]
```diff
@@ -15,10 +13,25 @@ TARGET=${TARGET-linux}
...
 libVFile="./libversion"
 
+function has_cmd {
+	command -v "$1" >/dev/null 2>&1
+}
+
+function calculate_sha256 {
+	if has_cmd "sha256sum"; then
+		command sha256sum "$1" | cut -f1 -d" "
+	elif has_cmd "shasum"; then
```
Comment (janisozaur) [inline install.sh]: This is available on Linux too, so you can make this the first check and only test for `sha256sum` if missing.

Comment (LinusU) [inline install.sh]: Sure thing :+1:

File: install.sh [hunk 3]
```diff
@@ -102,66 +115,46 @@ function install_local_libs {
...
+if [[ "$(uname)" == "Darwin" ]]; then
+	if ! has_cmd "brew"; then
+		echo "Homebrew is not installed, or brew is not in your \$PATH"
+		echo "install instructions: http://brew.sh/"
+		exit 1
+	fi
+
+	# Very possible I'm missing some dependencies here.
+	brew install cmake
+	brew install jansson sdl2 sdl2_ttf speex --universal
```
Comment (janisozaur) [inline install.sh]: these are not needed it `$TARGET == "windows"`

File: install.sh [hunk 4]
```diff
@@ -102,66 +115,46 @@ function install_local_libs {
...
+echo "TARGET = $TARGET"
+
+if [[ "$(uname)" == "Darwin" ]]; then
+	if ! has_cmd "brew"; then
+		echo "Homebrew is not installed, or brew is not in your \$PATH"
+		echo "install instructions: http://brew.sh/"
+		exit 1
+	fi
+
+	# Very possible I'm missing some dependencies here.
```
Comment (janisozaur) [inline install.sh]: I don't think it's the case, as was proven by successful builds. You can probably remove this comment now.

**Comment by janisozaur** [comment]
Can you perhaps update readme.md together with this commit?

I think it would worth mentioning that for now we only seem to support 10.11. Can you confirm that (as @kkirby reported) and mention in readme and perhaps add a check (together with a warning) in install/build script?

**Comment by janisozaur** [comment]
I just remembered I saw `-mmin-macosx-version` option to compiler (and similar one to linker) on OS X, do you think it would make sense to have this hard limit implemented?

File: install.sh (hunk 4 above)
Comment (LinusU) [inline install.sh]: Thought I removed this, will fix

**Comment by LinusU** [comment]
I've updated the readme as well, looks good?

Are we really sure that this doesn't run on any older OS X? I don't see a reason why it wouldn't work. Until we are sure I don't think we should put that in as a requirement.

**Comment by janisozaur** [comment]
@kkirby said he's getting this error on OS X 10.10, you may want to confirm yourself.

http://pastebin.com/zsP8z37U

**Comment by LinusU** [comment]
Hmm, that's an unfortunate stack trace :) it's going to be hard to debug...

Maybe we should only support 10.11 and up then. How would you like to implement it? readme, install.sh, etc.

**Comment by janisozaur** [comment]
I'm not OS X user myself, so I can only suggest things here.

Personally I would at least add check to cmake and display warning if OS X version is lower than 10.11.

The other thing would be either the mentioned hard limit (you can make it turn-off-able with some cmake switch like `FORCE_OLD_OSX`) or somewhere in `platform_init` (or another suitable function) display a message.

But please consult best practices and other users of your platform. Perhaps @IntelOrca can say something about it too.

**Comment by IntelOrca** [comment]
A yellow warning message seems sensible if you try to build it for a non-supported version.
And further information in readme.

**Comment by kkirby** [comment]
Maybe we should ask Reddit to test? I'm 100% open to the idea that my work laptop running 10.10 is botched. I'm also wondering if the retina display could be a factor. My 10.10 machine is retina and my 10.11 isn't. Though, the stack trace I get on my 10.10 machine doesn't lead me to believe this, I'm pulling at straws.

**Comment by LinusU** [comment]
My 10.11 also isn't retina. I think asking Reddit is a good idea :)

**Comment by kkirby** [comment]
I'll whip up a script later today so people can easily run a test case. Basically I'll just log the device info, run the game, and see if it crashes. If it does, I'll save the crash info. Then I'll auto-submit the results to a server.

**Comment by kkirby** [comment]
This is what I came up with for a script:

https://github.com/kkirby/OpenRCT2_reporter

Feedback would be greatly appreciated.

I modified openrct2 to exit with a status code of 2 after 100 loops in `platform_process_messages`. So if a status code of 2 isn't returned, then something went wrong.

I also added in dialogs for browsing to the rct2 install directory.

**Comment by kkirby** [comment]
Ha, wonderful. The addition of my `platform_show_messagebox` and `platform_open_directory_browser` causes the application to crash with a very similar stack trace on my 10.10 machine.

**Comment by kkirby** [comment]
OS X 10.10 is no longer an issue once my PR #2472 is committed. Apparently the argparse library inadvertently modifies the value of the `argv` variable passed in, and OS X reads from it in `AppKit` later down the line.

**Comment by LinusU** [comment]
Since we found the issue with older OS X and since it's being fixed, I don't think that we need to add an warning anymore.

@IntelOrca Is there anything more you want me to do with this? I think it would be very beneficiary for people that are trying to get OpenRCT2 running natively on OS X if we got this merged.

**Comment by LinusU** [comment]
Thanks!

**Pull request merged**